go-apps/meep-rnis/server/rnis.go

```diff
@@ -1426,126 +1487,151 @@ func rabRelSubscriptionsGET(w http.ResponseWriter, r *http.Request) {
 }
 
-func rabRelSubscriptionsPUT(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+func registerRr(rabRelSubscription *RabRelSubscription, subsIdStr string) {
+	subsId, _ := strconv.Atoi(subsIdStr)
+	mutex.Lock()
+	defer mutex.Unlock()
 
-	vars := mux.Vars(r)
-	subIdParamStr := vars["subscriptionId"]
-	var response InlineResponse2009
-	rabRelSubscription1 := new(RabRelSubscription1)
+	rrSubscriptionMap[subsId] = rabRelSubscription
+	if rabRelSubscription.ExpiryDeadline != nil {
+		//get current list of subscription meant to expire at this time
+		intList := subscriptionExpiryMap[int(rabRelSubscription.ExpiryDeadline.Seconds)]
+		intList = append(intList, subsId)
+		subscriptionExpiryMap[int(rabRelSubscription.ExpiryDeadline.Seconds)] = intList
+	}
+	log.Info("New registration: ", subsId, " type: ", rabRelSubscriptionType)
+}
 
-	decoder := json.NewDecoder(r.Body)
-	err := decoder.Decode(&rabRelSubscription1)
-	if err != nil {
-		log.Error(err.Error())
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
+func deregisterCc(subsIdStr string, mutexTaken bool) {
+	subsId, _ := strconv.Atoi(subsIdStr)
+	if !mutexTaken {
+		mutex.Lock()
+		defer mutex.Unlock()
 	}
+	ccSubscriptionMap[subsId] = nil
+	log.Info("Deregistration: ", subsId, " type: ", cellChangeSubscriptionType)
+}
 
-	rabRelSubscription := rabRelSubscription1.RabRelSubscription
+func deregisterRe(subsIdStr string, mutexTaken bool) {
+	subsId, _ := strconv.Atoi(subsIdStr)
+	if !mutexTaken {
+		mutex.Lock()
+		defer mutex.Unlock()
+	}
 
-	selfUrl := strings.Split(rabRelSubscription.Links.Self, "/")
-	subsIdStr := selfUrl[len(selfUrl)-1]
+	reSubscriptionMap[subsId] = nil
+	log.Info("Deregistration: ", subsId, " type: ", rabEstSubscriptionType)
+}
 
-	if subsIdStr != subIdParamStr {
-		http.Error(w, "Body content not matching parameter", http.StatusInternalServerError)
-		return
+func deregisterRr(subsIdStr string, mutexTaken bool) {
+	subsId, _ := strconv.Atoi(subsIdStr)
+	if !mutexTaken {
+		mutex.Lock()
+		defer mutex.Unlock()
 	}
 
-	if isSubscriptionIdRegisteredRr(subsIdStr) {
-		registerRr(rabRelSubscription, subsIdStr)
+	rrSubscriptionMap[subsId] = nil
+	log.Info("Deregistration: ", subsId, " type: ", rabRelSubscriptionType)
+}
 
-		_ = rc.JSONSetEntry(baseKey+rabRelSubscriptionType+":"+subsIdStr, ".", convertRabRelSubscriptionToJson(rabRelSubscription))
+func delSubscription(keyPrefix string, subsId string, mutexTaken bool) error {
 
-		response.RabRelSubscription = rabRelSubscription
-		jsonResponse, err := json.Marshal(response)
-		if err != nil {
-			log.Error(err.Error())
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-			return
-		}
-		w.WriteHeader(http.StatusOK)
-		fmt.Fprintf(w, string(jsonResponse))
-	} else {
-		w.WriteHeader(http.StatusNotFound)
-	}
+	err := rc.JSONDelEntry(keyPrefix+":"+subsId, ".")
+	deregisterCc(subsId, mutexTaken)
```

Comment:
Same comment as before... shouldn't we check which subscription type we are dealing with before removing the subscription ID from all maps?

Reply:
same comment as abvove...